Remove last modified dates from algorithm documentation pages

martyngigg opened the pull request:
**Description of work.**

The source link dates were parsed from the git log of the source
files and would include *any* change to the file regardless
of whether the change was meaningful, e.g. a formatting
change.
The preferred option going forward is for each page to
document any change in behaviour across versions in its
own changelog on the page.

<!-- If the original issue was raised by a user they should be named here. Do not leak email addresses
**Report to:** [user name]
-->

**To test:**

* Build the docs and check an algorithm page, e.g. Rebin and it should no longer have the  date is was modified.
* I would encourage running cmake with the following options to get the quickest docs build:
  * `DOCS_PLOTDIRECTIVE=OFF`
  * `DOCS_SCREENSHOTS=OFF`
  * `DOCS_DOTDIAGRAMS=OFF`
  * `DOCS_MATH_EXT=sphinx.ext.mathjax`

*There is no associated issue.*

<!-- Ensure the base of this PR is correct (e.g. release-next or master)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

martyngigg committed:
Remove last modified dates from algorithm docs pages

These dates were parsed from the git log of the source
files and would include *any* change to the file regardless
of whether the change was meaningful, e.g. a formatting
change.
The preferred option going forward is for each page to
document any change in behaviour across versions in its
own changelog on the page.

martyngigg committed:
Add release note for last-modified date removal

ConorMFinn commented on docs/source/release/v6.3.0/33349_framework.rst:
```diff
@@ -0,0 +1,10 @@
+  changes to the algorithm had actually occurred and could be something as
+  simple as formatting changes. Moving forward, each algorithm will now
+  document any significant changes of behaviour across versions in a dedicated
+  section within its own page.
```
Would it be worth mentioning here that we want to call that dedicated section the Changelog?

martyngigg replied:
I think at this point that's probably too much detail. The contents on each page should make it obvious I think.

cailafinn reviewed (changes requested):
Just one minor note on the notes. 
Also just want to check: should the "last updated" note for the doc at the very bottom of the page be staying?

ConorMFinn commented:
> > Just one minor note on the notes. Also just want to check: should the "last updated" note for the doc at the very bottom of the page be staying?
> 
> Sorry Conor, I don't quite understand what you mean - could you help me?

This bit
![Screenshot 2022-01-21 at 09 45 42](https://user-images.githubusercontent.com/47181718/150505022-e037db9b-b25e-4eb1-b3ee-6f4de2a14ff1.png)

martyngigg commented:
Ah thank you - I don't know how I didn't spot that. That must be a Sphinx thing but I agree that it's not useful in this context as we do a clean build every time we deploy so it will always say yesterday's build time and is more misleading. I'll see if I can find a switch to turn it off.
Would you be happy if that went in a separate pull request as it'll be in a separate part of the code?

martyngigg commented:
Actually this looks trivial to do so I'll include it here.

martyngigg committed:
Remove page last modified date from footer

For the same reason as the algorithm modified
time this can be misleading as we always clean
build the docs so the latest build time will
always be displayed

cailafinn reviewed (approved):
Good to have removed this, looks good to me! 👍
